otlp-logger.js

```diff
@@ -0,0 +1,153 @@
+'use strict'
+
+const {
+  LoggerProvider,
+  SimpleLogRecordProcessor,
+  BatchLogRecordProcessor
+} = require('@opentelemetry/sdk-logs')
+
+const { SeverityNumber } = require('@opentelemetry/api-logs')
+const { logs } = require('@opentelemetry/api-logs')
+const { Resource } = require('@opentelemetry/resources')
+
+const DEFAULT_MESSAGE_KEY = 'msg'
+
+/**
+ * @typedef {Object} Options
+ * @property {string} loggerName
+ * @property {string} serviceName
+ * @property {string} serviceVersion
+ * @property {boolean} includeTraceContext
+ * @property {import('@opentelemetry/sdk-logs').LogRecordExporter} [logRecordExporter]
+ * @property {boolean} [useBatchProcessor=true]
+ * @property {string} [messageKey="msg"]
+ *
+ * @param {Options} opts
+ */
+function getOtlpLogger (opts) {
+  const loggerProvider = new LoggerProvider({
+    resource: new Resource({
+      'service.name': opts.serviceName,
+      'service.version': opts.serviceVersion
```

Comment:
```suggestion
      'service.version': opts.serviceVersion,
      ...opts.resourceAttributes
```

Can you add an option for additional resource attributes? I feel this is a must-have

Reply:
It's my experience with the OTel node SDK is that this environment variable isn't used automatically. I've had to parse it myself and pass it in. Does yours differ?

Reply:
There are also attributes that you may want to add that would not be best set in an environment variable. e.g...

```js
{ [SemanticResourceAttributes.SERVICE_INSTANCE_ID]: `${os.hostname()}:${process.pid}` }
```

Reply:
> It's my experience with the OTel node SDK is that this environment variable isn't used automatically. I've had to parse it myself and pass it in. Does yours differ?

It is possible to do that. You are correct this does not work in the current implementation. I should add a ResourceDetector to read the attributes.

> There are also attributes that you may want to add that would not be best set in an environment variable. e.g...
> 
> ```js
> { [SemanticResourceAttributes.SERVICE_INSTANCE_ID]: `${os.hostname()}:${process.pid}` }
> ```

Good point. We should expand the API for that kind of use cases.

Reply:
https://github.com/Vunovati/pino-opentelemetry-transport/issues/4